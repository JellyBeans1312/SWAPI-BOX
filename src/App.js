import React, { Component } from 'react';
import './App.css';
import OpeningCrawl from './OpeningCrawl';
import CardContainer from './CardContainer';

class App extends Component {
  constructor() {
    super();
    this.state = {
    filmData: [],
    peopleData: [],
    planetData: [],
    vehicleData: [],
    error: ''
    }
  }

  componentDidMount() {
    // fetch('https://swapi.co/api/films')
    // .then(response => response.json())
    // .then(data => this.setState({
    //   filmData: data.results[0]
    // }))
    // .catch(error => this.setState({ error }))

    // fetch('https://swapi.co/api/people')
    // .then(response => response.json())
    // .then(data => this.setState({
    //   peopleData: data.results
    // }))
    // .catch(error => this.setState({ error }))

    // fetch('https://swapi.co/api/planets')
    // .then(response => response.json())
    // .then(data => this.setState({
    //   planetData: data.results
    // }))
    // .catch(error => this.setState({ error }))

    // fetch('https://swapi.co/api/vehicles')
    // .then(response => response.json())
    // .then(data => this.setState({
    //   vehicleData: data.results
    // }))
    // .catch(error => this.setState({ error }))
  }

  render () {
   const { filmData, peopleData, planetData, vehicleData } = this.state;
    return (
      <main>
        <OpeningCrawl 
        title={filmData.title} 
        date={filmData.release_date}
        episode={filmData.episode_id}
        text={filmData.opening_crawl}  />
        <section>
          <h1>SWAPI-BOX</h1>
          <button className='favorites' img="">Favs <span className="favsCount" >0</span>
          </button>
        </section>
        <nav>
          <button>People</button>
          <button>Planets</button>
          <button>Vehicles</button>
        </nav>
        <section className="card-container">
          <CardContainer />
        </section>
      </main>
      )
  }  
}

export default App;
