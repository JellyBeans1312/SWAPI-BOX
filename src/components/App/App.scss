.App {
  text-align: center;
}

body {
  background: url('https://wallpaperplay.com/walls/full/c/6/4/17373.jpg');
  background-size: cover;
  background-attachment: fixed;
  position: absolute;
  width: 100vw;
  height: 100%;
}
