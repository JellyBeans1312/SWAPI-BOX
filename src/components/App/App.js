import React, { Component } from 'react';
import './App.scss';
import OpeningCrawl from '../OpeningCrawl/OpeningCrawl';
import CardContainer from '../CardContainer/CardContainer';

class App extends Component {
  constructor() {
    super();
    this.state = {
    filmData: [],
    peopleData: [],
    planetData: [],
    vehicleData: [],
    favoriteCards: [],
    landingPage: true,
    error: ''
    }
  }

  hideLanding = () => {
    this.setState({ landingPage: false })
  }

  showLanding = () => {
    this.setState({ landingPage: true })
  }

  componentDidMount() { 
    // const number =  Math.floor(Math.random() * (6 - 2 + 1)) + 1
    // fetch('https://swapi.co/api/films')
    // .then(response => response.json())
    // .then(data => this.setState({
    //   filmData: data.results[number]
    // }))
    // .catch(error => this.setState({ error }))

    // fetch('https://swapi.co/api/people/')
    //  .then(response => response.json())
    //  .then(data => this.fetchPeople(data.results))
    //  .then(data => this.fetchSpecies(data))
    //  .then(data => data.map( datum => {
    //     const clean = [
    //       datum.name,
    //       `Homeworld: ${datum.homeworld}`,
    //       `Species: ${datum.species}`,
    //       `Population: ${datum.population}`,
    //       null,
    //       datum.id
    //     ]
    //     return clean
    //  }))
    //  .then(peopleData => this.setState({ peopleData }))
    //  .catch(error => this.setState({ error }))

    fetch('https://swapi.co/api/planets/')
    .then(response => response.json())
    .then(data => this.fetchResidents(data.results))
    .then(data => data.map(datum => {
      console.log(datum)
          const clean = [
            datum.name,
            `Terrain: ${datum.terrain}`,
            `Population: ${datum.population}`,
            `Climate: ${datum.climate}`,
            `Residents: ${datum.residents}`,
          ]
          return clean
    }))
    .then(planetData => this.setState({ planetData }))
    .catch(error => this.setState({ error }))

    // fetch('https://swapi.co/api/vehicles')
    // .then(response => response.json())
    // .then(data => this.setState({
    //   vehicleData: data.results.map(vehicle => {
    //     const clean = [
    //       vehicle.name,
    //       `Model: ${vehicle.model}`,
    //       `Class: ${vehicle.vehicle_class}`,
    //       `Passengers: ${vehicle.passengers}`, 
    //       null,
    //       vehicle.created
    //     ]
    //     return clean
    //   })
    // }))
    // .catch(error => this.setState({ error }))
  }

//   fetchPeople = (people) => {
//     const promises = people.map(person => {
//       return fetch(person.homeworld)
//         .then(response => response.json())
//         .then(data => ({
//           homeworld: data.name,
//           population: data.population,
//           name: person.name,
//           species: person.species
//          }))
//         .catch(error => console.log(error));
//     });
//     return Promise.all(promises);
//   };


//  fetchSpecies = (people) => {
//   const promises = people.map(person => {
//     return fetch(person.species)
//       .then(response => response.json())
//       .then(data => ({
//         ...person,
//         id: person.created,
//         species: data.name
//       }))
//       .catch(error => console.log(error));
//   });
//   return Promise.all(promises);
// };

fetchResidents = (planets) => {
  const allPlanets = planets.map(planet => {
    let res = [];
    planet.residents.map(resident => {
      return fetch(resident)
      .then(response => response.json())
      .then(data => res.push(data.name))
      .catch(error => this.setState({ error }))
    });
    console.log(res)
    return {
      name: planet.name,
      terrain: planet.terrain,
      population: planet.population,
      climate: planet.climate,
      residents: res
    }
  });
  return allPlanets
};

  render () {
   const { filmData, peopleData, planetData, vehicleData, favoriteCards, landingPage} = this.state;
    return (
    
      <main>
        {landingPage && 
          <OpeningCrawl 
          title={filmData.title} 
          date={filmData.release_date}
          episode={filmData.episode_id}
          text={filmData.opening_crawl} 
          hideLanding={this.hideLanding} />
        }
        {!landingPage && 
        <>
        <section className='idk-change-later'>
        <button className='show-movie star-wars-text' onClick={this.showLanding}>
          Show Movie Stuff
        </button>
          <h1 className='star-wars-text header'>SWAPI-BOX</h1>
          <button className='favorites star-wars-text'>
            Favorites <span> {favoriteCards.length}</span>
          </button>
        </section>
        <nav>
          <button className='star-wars-text nav_button'>People</button>
          <button className='star-wars-text nav_button'>Planets</button>
          <button className='star-wars-text nav_button'>Vehicles</button>
        </nav>
        <section>
          {/* if people is clicked           */}
          {/* <CardContainer data={peopleData}/> */}
          
          {/* if planets is clicked */}
          <CardContainer data={planetData}/>

          
          {/* if vehicles is clicked */}
          {/* <CardContainer data={vehicleData}/> */}


        </section>
        </>
        }
      </main>
      )
  }  
}

export default App;
