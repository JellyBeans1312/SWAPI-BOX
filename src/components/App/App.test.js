import React from 'react';
import App from './App';
import { shallow } from 'enzyme'
// import { MemoryRouter } from 'react-router-dom';
// import { Home } from '../Home/Home'
// import CardContainer from '../CardContainer/CardContainer';
// import { NotFound } from '../NotFound/NotFound';


describe ('App', () => {
  let wrapper;
  const mockFavorite = [
    'aint',
    'no',
    'thang',
    'but a',
    'chicken',
    'wang',
    false
  ];

  const mockMovie = {
    title: 'aint no thang',
    release_date: 'year',
    episode: 'four',
    text: 'but a chicken wang',
  }

  beforeEach(() => {
    wrapper = shallow(<App />)
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should hide landing page when button is clicked', () => {
    const mockEvent = { target: {name: 'landingPage', value: false}};
    const expectedValue = false;

    wrapper.instance().hideLanding(mockEvent);

    expect(wrapper.state('landingPage')).toEqual(expectedValue);
  });

  it('should show landing page when button is clicked', () => {
    const mockEvent = { target: {name: 'landingPage', value: true}};
    const expectedValue = true;

    wrapper.instance().showLanding(mockEvent);

    expect(wrapper.state('landingPage')).toEqual(expectedValue);
  });

  it('should add card to favorite card array when clicked', () => {
    expect(wrapper.state('favoriteCards').length).toEqual(0)
    wrapper.instance().updateFavoriteCard(mockFavorite)
    expect(wrapper.state('favoriteCards')).toEqual([mockFavorite])
    wrapper.instance().updateFavoriteCard(mockFavorite)
    expect(wrapper.state('favoriteCards').length).toEqual(0)
  });

  it('should set movie to state when setMovie is called', () => {
    expect(wrapper.state('filmData')).toEqual([])
    wrapper.instance().setMovie(mockMovie)
    expect(wrapper.state('filmData')).toEqual(mockMovie)
  })

  // describe('Routes', () => {
  //   it('should route to the home page', () => {
  //     const wrapper = mount(
  //       <MemoryRouter initialEntries={['/']} initialIndex={0}>
  //         <App />
  //       </MemoryRouter>
  //     );
  
  //     expect(wrapper.find(Home)).toHaveLength(1);
  //   });

  //   it('should route to people page', () => {
  //     const wrapper = mount(
  //       <MemoryRouter initialEntries={['/people']} initialIndex={0}>
  //         <App />
  //       </MemoryRouter>
  //     );
  
  //     expect(wrapper.find(CardContainer)).toHaveLength(1);
  //   });

  //   it('should route to the planets page', () => {
  //     const wrapper = mount(
  //       <MemoryRouter initialEntries={['/planets']} initialIndex={0}>
  //         <App />
  //       </MemoryRouter>
  //     );
  
  //     expect(wrapper.find(CardContainer)).toHaveLength(1);
  //   });

  //   it('should route to the vehicles page', () => {
  //     const wrapper = mount(
  //       <MemoryRouter initialEntries={['/vehicles']} initialIndex={0}>
  //         <App />
  //       </MemoryRouter>
  //     );
  
  //     expect(wrapper.find(CardContainer)).toHaveLength(1);
  //   });

  //   it('should route to the favorites page', () => {
  //     const wrapper = mount(
  //       <MemoryRouter initialEntries={['/favorites']}>
  //         <App />
  //       </MemoryRouter>
  //     );
  
  //     expect(wrapper.find(CardContainer)).toHaveLength(1);
  //   });

  //   it('should route to the 404 page', () => {
  //     const wrapper = mount(
  //       <MemoryRouter initialEntries={['/sometjing']}>
  //         <App />
  //       </MemoryRouter>
  //     );
  
  //     expect(wrapper.find(NotFound)).toHaveLength(1);
  //   });
  // });
});

