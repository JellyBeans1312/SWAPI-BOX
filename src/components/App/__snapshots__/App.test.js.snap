// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App sholud match a snapshot 1`] = `
<main>
  <OpeningCrawl
    hideLanding={[Function]}
  />
</main>
`;

exports[`App should match a snapshot 1`] = `
<main>
  <OpeningCrawl
    hideLanding={[Function]}
  />
</main>
`;
